doc/nrf/gs_testing.rst

```diff
@@ -120,3 +120,93 @@ To connect to the nRF9160-based kit with LTE Link Monitor, perform the following
    .. note::
 
       In the case of nRF9160 DK, the reset button must be pressed to restart the device and to start the application.
+
+.. _gs_debugging:
+
+Debugging an application
+************************
+
+To debug an application, set up the debug session as described in `Debugging an application`_ in the |nRFVSC| documentation.
+
+If you use a multi-core SoC, for example from the nRF53 Series, you need a single debug session to debug the application core firmware.
+To debug the firmware running on the network core, you need to set up two separate debug sessions: one for the network core and one for the application core.
+When debugging the network core, the application core debug session runs in the background and you can debug both cores if needed.
+
+Complete the following steps to start debugging the network core:
+
+1. Set up sessions for the application core and network core as mentioned in `Debugging an application`_.
+#. Select the appropriate CPU for debugging in each session, corresponding to the application core and the network core of your SoC, respectively.
+#. Once both sessions are established, execute the code on the application core.
+
+   The startup code releases the ``NETWORK.FORCEOFF`` signal to start the network core and allocates the necessary GPIO pins for it.
+#. Start code execution on the network core in the other debug session.
+
+If you want to reset the network core while debugging, make sure to first reset the application core and execute the code.
+
+.. note::
+  Debugging firmware on the application core in the non-secure domain (that is, when using an ``_ns`` build target), is currently not supported.
+
+Debug configuration
+===================
+
+The following Kconfig options are used by the |VSC| debugger in the |nRFVSC|:
+
+* :kconfig:option:`CONFIG_DEBUG_OPTIMIZATIONS` set to ``y``
```

Comment:
I have not used this VSC debugger but does it really enable this config options automatically? That would be problematic... Adding DEBUG_OPTIMIZATIONS is needed only if you want to have correct C code stepping and you don't care about it if you are fine with assembler.
Forcefully enabling that might impact application sizes and cause some apps/samples not to build.

Or is it that it is suggested to enable these options to have better debugging experience?

Reply:
@trond-snekvik , how does this look from the extension perspective? The VSC ext docs don't state these are enabled by default, but they also use wording that might be ambiguous ("**If you forgot** to enable debugging during the build configuration step, add the following lines into the application configuration file [...]").

Reply:
I hope this is just bad wording in the guide, but until we know what is correct could @pdunaj block the PR, so that it doesn't get merged until this is resolved ?

Reply:
Put DNM on the PR for now :) I'll check with Trond today.

Reply:
Is `Enable debug configuration` a setting that the user selects independently of launching the debugger itself ?
Or is it automatically selected by the debugger.

I hope it's an independent setting that user can select independently of the debugger.
If that's the case, then the wording should be updated to reflect that selection of the setting could improve the debugger session, but that it is not required for debugging.

Reply:
As @tejlmand mentioned, it's not a problem as long as user can control this. I assume this wording will be fixed.

Reply:
The user can indeed enable or disable this manually, yes.

Reply:
> (We're planning to update the wording in this task for the VSC extension: https://projecttools.nordicsemi.no/jira/browse/VSC-978 )

I see no mentioning of this PR or updating of the wording, so until it is clear in the task what must be done wrt. wording then I cannot give a +1, cause there's too high a risk that the wording will not be updated if not explicitly stated.